bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/discovery/DanfossAirUnitDiscoveryService.java

```diff
@@ -0,0 +1,139 @@
+        logger.debug("Try to discover all Danfoss Air CCM devices");
+
+        try (DatagramSocket socket = new DatagramSocket()) {
+
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface == null || networkInterface.isLoopback() || !networkInterface.isUp()) {
+                    continue;
+                }
+                for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                    if (interfaceAddress == null || interfaceAddress.getBroadcast() == null) {
+                        continue;
+                    }
+                    logger.debug("Sending broadcast on interface {} to discover Danfoss Air CCM device...",
+                            interfaceAddress.getAddress().toString());
+                    sendBroadcastToDiscoverThing(socket, interfaceAddress.getBroadcast());
+                }
+            }
+
+        } catch (IOException e) {
+            logger.debug("No Danfoss Air CCM device found. Diagnostic: {}", e.getMessage());
+        }
+    }
+
+    private void sendBroadcastToDiscoverThing(DatagramSocket socket, InetAddress broadcastAddress) throws IOException {
+        socket.setBroadcast(true);
+        socket.setSoTimeout(500);
+        // send discover
+        byte[] sendBuffer = DISCOVER_SEND;
+        DatagramPacket sendPacket = new DatagramPacket(sendBuffer, sendBuffer.length, broadcastAddress, BROADCAST_PORT);
+        socket.send(sendPacket);
+        logger.debug("Discover message sent");
+
+        // wait for responses
+        while (true) {
+            byte[] receiveBuffer = new byte[7];
+            DatagramPacket receivePacket = new DatagramPacket(receiveBuffer, receiveBuffer.length);
+            try {
+                socket.receive(receivePacket);
+            } catch (SocketTimeoutException e) {
+                break; // leave the endless loop
+            }
+
+            byte[] data = receivePacket.getData();
+            if (Arrays.equals(data, DISCOVER_RECEIVE)) {
+                logger.debug("Discover received correct response");
+
+                String host = receivePacket.getAddress().getHostName();
+                Map<String, Object> properties = new HashMap<>();
+                properties.put("host", host);
+
+                logger.debug("Adding a new Danfoss Air Unit CCM '{}' to inbox", host);
+
+                ThingUID uid = new ThingUID(THING_TYPE_AIRUNIT, String.valueOf(receivePacket.getAddress().hashCode()));
+
+                DiscoveryResult result = DiscoveryResultBuilder.create(uid).withRepresentationProperty("host")
+                        .withProperties(properties).withLabel("Danfoss HRV").build();
+                thingDiscovered(result);
+
+                logger.debug("Thing discovered '{}'", result);
```

Comment:
Is this missing a break, because if the device responds it will just keep running or will it always throw an `SocketTimeoutException`?

Reply:
It will process packets as long as there is anything to read, otherwise it'll time out and break the loop.